### PR TITLE
Creation of a command to delete outdated notifications

### PR DESCRIPTION
**Pull request by Nath82**
#100 After several discussions we decided that the best way to deal with this problem in the long term was to create a Django command to remove old notifications. The default value is 30 days, which considerably increases query execution speed.

### CODE REVIEW

File: src/nrc/datamodel/management/commands/clean_old_notifications.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+class Command(BaseCommand):
+    def handle(self, **options):
+        date_limite = timezone.now() - timedelta(
+            days=settings.NOTIFICATION_NUMBER_OF_DAYS_RETAINED
+        )
+        notifications_filtrees = (
+            Notificatie.objects.annotate(
+                extracted_date=F("forwarded_msg__aanmaakdatum")
+            )
+            .filter(extracted_date__lt=date_limite)
```
Comment (annashamray) [inline src/nrc/datamodel/management/commands/clean_old_notifications.py:20]: I'm not sure why do you need `annotate` here

```suggestion
            Notificatie.objects.filter(forwarded_msg__aanmaakdatum__lt=date_limite)
```

File: src/nrc/datamodel/management/commands/clean_old_notifications.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+from django.core.management.base import BaseCommand
+from django.db.models import F
+from django.utils import timezone
+
+from ...models import Notificatie
+
+
+class Command(BaseCommand):
+    def handle(self, **options):
+        date_limite = timezone.now() - timedelta(
```
Comment (annashamray) [inline src/nrc/datamodel/management/commands/clean_old_notifications.py:13]: Could you please create a test for the command?

**Review by annashamray — CHANGES_REQUESTED** [review]
@alextreme @Nath82 
We already have celery in ON. Do you want to introduce celery beat and make it a scheduled task instead?

File: src/nrc/datamodel/management/commands/clean_old_notifications.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+        notifications_filtered = Notificatie.objects.filter(
+            forwarded_msg__aanmaakdatum__lt=date_limit
+        ).delete()
+
+        self.stdout.write(
+            str(notifications_filtered[0])
+            + " notifications have been deleted : "
+            + str(notifications_filtered[1])
+        )
```
Comment (annashamray) [inline src/nrc/datamodel/management/commands/clean_old_notifications.py:24]:
```suggestion
        self.stdout.write(f"{notifications_filtered[0]} notifications have been deleted")
```

I think the number of deleted notifications is enough, the dictionary is no needed to display

File: src/nrc/tests/commands/test_clean_old_notifications.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        self.assertEqual(Notificatie.objects.count(), 1)
+        self.assertEqual(NotificatieResponse.objects.count(), 1)
+
+        self.assertEqual(Notificatie.objects.last(), self.notif_in_month)
+        self.assertEqual(
+            NotificatieResponse.objects.last(), self.notif_response_in_month
+        )
+
+    def test_change_of_period(self):
+        settings.NOTIFICATION_NUMBER_OF_DAYS_RETAINED = 60
```
Comment (annashamray) [inline src/nrc/tests/commands/test_clean_old_notifications.py:58]: if you want to change a django setting for a test, you can use `override_settings` decorator

**Review by annashamray — COMMENTED** [review]
I don't think making a task has a lot of sense if it's not called anywhere. 
@joeribekker @alextreme @Nath82 Is there a plan to introduce celery beat here?

File: src/nrc/datamodel/management/commands/clean_old_notifications.py (hunk 3 above)
Comment (Nath82) [reply src/nrc/datamodel/management/commands/clean_old_notifications.py:24]: This is the result with the dictionary :
image
It seems relevant to me, to know for NotificatieResponse

**Comment by joeribekker** [comment]
@annashamray I think celery beat should indeed be introduced.

File: src/nrc/conf/includes/base.py [hunk 5]
```diff
@@ -512,6 +520,13 @@
 MOZILLA_DJANGO_OIDC_DB_CACHE = "oidc"
 MOZILLA_DJANGO_OIDC_DB_CACHE_TIMEOUT = 5 * 60
 
+#
+# Delete Notifications
+#
+NOTIFICATION_NUMBER_OF_DAYS_RETAINED = config(
```
Comment (annashamray) [inline src/nrc/conf/includes/base.py:526]: please document this variable in `docs/configuration/env_config.md` file

File: src/nrc/conf/includes/base.py [hunk 6]
```diff
@@ -478,6 +479,13 @@
 # Celery
 CELERY_BROKER_URL = config("CELERY_BROKER_URL", "amqp://127.0.0.1:5672//")
 CELERY_RESULT_BACKEND = config("CELERY_RESULT_BACKEND", "redis://localhost:6379/1")
+CELERY_BEAT_SCHEDULE = {
```
Comment (annashamray) [inline src/nrc/conf/includes/base.py:482]: Please provide also tools for developers to run celery beat:
* bash file to run celery beat
* celery beat container in docker-compose.yml

**Review by annashamray — CHANGES_REQUESTED** [review]

File: docker-compose.yml [hunk 7]
```diff
@@ -38,6 +38,16 @@ services:
       - rabbitmq
       - redis
 
+  celery-beat:
```
Comment (annashamray) [inline docker-compose.yml:41]: when I try to use docker-compose I have an error:


`ERROR: for open-notificaties_celery-beat_1  Cannot start service celery-beat: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "/celery_beat.sh": stat /celery_beat.sh: no such file or directory: unknown
`

**Review by annashamray — APPROVED** [review]

**Pull request merged**